src/error.rs

```diff
@@ -51,6 +51,10 @@ pub enum Error {
     #[error("Invalid chroot base path specified")]
     InvalidChrootBasePath,
 
+    /// Firecracker socket API error
+    #[error("Firecracker API call failed: {0}")]
+    FirecrackerAPIError(String),
```

Comment:
* We should also encode the status.
* It should be `Option<String>` so when the body is empty, it's `None`.